extide/gradle/netbeans-gradle-tooling/src/main/groovy/org/netbeans/modules/gradle/tooling/NbProjectInfoBuilder.groovy

```diff
@@ -341,10 +374,18 @@ class NbProjectInfoBuilder {
         }
         //visibleConfigurations = visibleConfigurations.findAll() { resolvable(it) }
         visibleConfigurations.each() {
-            def componentIds = []
+            def componentIds = new HashSet()
             def unresolvedIds = []
             def projectNames = []
             long time_inspect_conf = System.currentTimeMillis()
+            
+            it.dependencies.withType(ModuleDependency) {
+                def group = it.group != null ? it.group : '';
+                def name = it.name
+                def version = it.version != null ? ':' + it.version : ''
+                def id = group + ':' + name + version;
+                componentIds.add(id)                
```

Comment:
is it OK to pass on malformed component IDs, e.g. no name or no group ?

Reply:
No group, no version names are valid.